src/main/scala/io/iohk/ethereum/ledger/BlockRewardCalculator.scala

```diff
@@ -56,13 +53,31 @@ class BlockRewardCalculator(config: MonetaryPolicyConfig) {
     if (era == 0) {
       val numer = firstEraOmmerMiningRewardMaxNumer - (blockNumber - ommerNumber - 1)
       (firstEraBlockReward * numer) / firstEraOmmerMiningRewardDenom
-    } else {
-      val eraMultiplier = rewardReductionRateNumer.pow(era)
-      val eraDivisor = rewardReductionRateDenom.pow(era)
-      (firstEraBlockReward * ommerMiningRewardNumer * eraMultiplier) / (ommerMiningRewardDenom * eraDivisor)
-    }
+    } else
+      calcMinerBaseReward(era) * ommerMiningRewardNumer / ommerMiningRewardDenom
```

Comment:
You're multiplying the result of division, thus introducing rounding errors

Reply:
Yes, this could cause rounding error, however as this seems to be the more closer approach to how it is [calculated in Geth](https://github.com/ethereumproject/go-ethereum/pull/220/files#diff-5b5942d1b7feac47d38b879796555098R197) and `ommerMiningRewardNumer=1`, I decided to leave it this way.
Do you think it would be better to change this?

Reply:
No, if geth does it then it's fine. I thought about it and it's not devoid of logic to have it rounded down prior to another division.